metagraph/plugins/networkx/algorithms.py

```diff
@@ -147,9 +164,41 @@ def nx_betweenness_centrality(
             targets=targets,
             normalized=normalize,
             weight=graph.edge_weight_label,
-            # endpoints=include_endpoints,
         )
-        return PythonNodeMap(node_to_score_map,)
+        return PythonNodeMap(node_to_score_map)
+
+    @concrete_algorithm("centrality.closeness")
+    def nx_closeness_centrality(
+        graph: NetworkXGraph, nodes: mg.Optional[PythonNodeSet],
+    ) -> PythonNodeMap:
+        if nodes is None:
+            result = nx.closeness_centrality(
+                graph.value, distance=graph.edge_weight_label
+            )
+        else:
+            result = {
+                node: nx.closeness_centrality(
+                    graph.value, node, distance=graph.edge_weight_label
+                )
+                for node in nodes.value
+            }
+        return PythonNodeMap(result)
+
+    @concrete_algorithm("centrality.eigenvector")
+    def nx_eigenvector_centrality(
+        graph: NetworkXGraph, maxiter: bool = 100, tol: float = 1e-6
+    ) -> PythonNodeMap:
+        result = nx.eigenvector_centrality(
+            graph.value, maxiter, tol, weight=graph.edge_weight_label
+        )
+        return PythonNodeMap(result)
+
+    @concrete_algorithm("centrality.hits")
+    def nx_hits_centrality(
+        graph: NetworkXGraph, max_iter: int, tol: float, normalize: bool,
+    ) -> Tuple[PythonNodeMap, PythonNodeMap]:
+        hubs, authority = nx.hits(graph.value, max_iter, tol, normalized=normalize)
+        return PythonNodeMap(hubs), PythonNodeMap(authority)
```

Comment:
Is there any particular reason that `max_iter` is not a backend-specific parameter for `centrality.hits` but is for `centrality.eigenvector`? They have similar descriptions on the `NetworkX` documentation.

Reply:
iGraph has several implementations which don't take maxiter or tolerance and instead rely on LAPACK to perform the computation. I think these are common enough to include them even if a specific implementation decides to ignore them.